File: functions/getWebhooks.js
const needle = require('needle'), config = require('../config.json')

const webhooks = async(id) => {
    var hooks = (await needle("get", `https://discord.com/api/v9/channels/${id}/webhooks`, {
        headers: {
            Authorization: config.token
        }
    })).body
    return hooks
}

module.exports = {
    webhooks
}

File: nim-modules/getWebhooks.nim
import
    std/[httpclient, os, strformat]

proc getWebhooks*: string =
    var
        client = newHttpClient()
        params = commandLineParams()
    client.headers = newHttpHeaders({"Authorization": params[0], "Content-Type": "application/json"})
    let
        res = client.request(fmt"https://discord.com/api/v9/channels/{params[1]}/webhooks", HttpGet)
    return res.body

when isMainModule:
    echo getWebhooks()